lib/command/run.rb

```diff
@@ -48,80 +63,178 @@ class Run < Base
 
+      # Set cron job props
+      spec["type"] = "cron"
+      spec["job"] = { "schedule" => "* * * * *", "restartPolicy" => "Never" }
+
+      # Create workload clone
+      cp.apply_hash("kind" => "workload", "name" => workload_clone, "spec" => spec)
+    end
+    
+    def update_workload_clone
+      # Get base specs of workload
+      spec = cp.fetch_workload!(workload_clone).fetch("spec")
+      container_spec = spec["containers"].detect { _1["name"] == workload_to_clone } || spec["containers"].first
+      @container = container_spec["name"]
+
       # Override image if specified
       image = config.options[:image]
       image = latest_image if image == "latest"
       container_spec["image"] = "/org/#{config.org}/image/#{image}" if image
 
-      # Set runner
-      container_spec["env"] ||= []
-      container_spec["env"] << { "name" => "CONTROLPLANE_RUNNER", "value" => runner_script }
+      # Container overrides
+      container_spec["cpu"] = config.options[:cpu] if config.options[:cpu]
+      container_spec["memory"] = config.options[:memory] if config.options[:memory]
 
-      if config.options["use_local_token"]
-        container_spec["env"] << { "name" => "CONTROLPLANE_TOKEN",
-                                   "value" => ControlplaneApiDirect.new.api_token[:token] }
-      end
-
-      # Create workload clone
+      # Update workload clone
       cp.apply_hash("kind" => "workload", "name" => workload_clone, "spec" => spec)
     end
 
-    def runner_script # rubocop:disable Metrics/MethodLength
-      script = Scripts.helpers_cleanup
+    def build_data
+      data = <<~DATA
+        name: #{container}
+      DATA
+
+      if config.options[:use_local_token]
+        data += <<~ENV
+          env:
+            - name: CPLN_TOKEN
+              value: "#{ControlplaneApiDirect.new.api_token[:token]}"
+        ENV
+      end
+
+      command = nil
+      if config.options[:entrypoint]
+        if config.options[:entrypoint] == "none"
+          # Replaces the existing Dockerfile entrypoint with `bash`
+          command = "bash"
+        else
+          # Replaces the existing Dockerfile entrypoint with the provided option
+          command = config.options[:entrypoint]
+        end
+      end
 
-      if config.options["use_local_token"]
-        script += <<~SHELL
-          CPLN_TOKEN=$CONTROLPLANE_TOKEN
-          unset CONTROLPLANE_TOKEN
-        SHELL
+      if interactive
+        data += <<~COMMAND
+          #{"command: #{command}" if command}
+          args:
+            #{'- "bash"' unless command == "bash"}
+            - "-c"
+            - 'eval "tail -f /dev/null"'
+          COMMAND
+        @command = %(bash -c 'eval "#{args_join(config.args)}"')
+      else
+        data += <<~COMMAND
+          #{"command: #{command}" if command}
+          args:
+            #{'- "bash"' unless command == "bash"}
+            - "-c"
+            - 'eval "#{args_join(config.args)}"'
```

Comment:
This `eval` needs explaining.

Reply:
ask ai for help

Reply:
`eval` makes it easier to run multiline scripts and circumvent some path issues. Since this works, how about we leave it like this for now and investigate potentially removing `eval` in a follow up PR?